File: test_archives/py_functions/batch5/function4.py
class function4:
    
    def __init__(self, name, age, gender, income):
        self.name = name
        self.age = age
        self.gender = gender
        self.income = income
        self.list1 = []
        self.list2 = []
        self.dict1 = {}
        self.dict2 = {}
        self.str1 = ""
        self.str2 = ""
        self.int1 = 0
        self.int2 = 0
        self.float1 = 0.0
        self.float2 = 0.0
        self.char1 = ''
        self.char2 = ''
        
    def calculate_final_value(self):
        final_value = self.int1 + self.int2 + float(self.float1) + float(self.float2) + len(self.list1) + len(self.list2) + len(self.dict1) + len(self.dict2) + len(self.str1) + len(self.str2) + ord(self.char1) + ord(self.char2)
        return final_value

File: test_archives/cpp_functions/batch5/function4.cpp
#include <iostream>
#include <string>
#include <vector>
#include <map>
using namespace std;

class function4{
public:
    string name;
    int age;
    string gender;
    int income;
    vector<int> list1;
    vector<int> list2;
    map<string, int> dict1;
    map<string, int> dict2;
    string str1;
    string str2;
    int int1;
    int int2;
    float float1;
    float float2;
    char char1;
    char char2;

    function4(string name, int age, string gender, int income){
        this->name = name;
        this->age = age;
        this->gender = gender;
        this->income = income;
    }

    int calculate_final_value(){
        int final_value = this->int1 + this->int2 + float(this->float1) + float(this->float2) + this->list1.size() + this->list2.size() + this->dict1.size() + this->dict2.size() + this->str1.size() + this->str2.size() + int(this->char1) + int(this->char2);
        return final_value;
    }
};